src/Support/Utility.php

```diff
@@ -399,30 +399,81 @@ public static function contentstackRequest($queryObject = '', $type = '')
             if ($Headers["branch"] !== '' && $Headers["branch"] !== "undefined") {
                 $request_headers[] = 'branch: '.$Headers["branch"];
             }
+
+            $proxy_details = $queryObject->contentType->stack->proxy;
+            $timeout = $queryObject->contentType->stack->timeout;
+
             curl_setopt($http, CURLOPT_HTTPHEADER, $request_headers);
             
             curl_setopt($http, CURLOPT_HEADER, false);
             // setting the GET request
             curl_setopt($http, CURLOPT_CUSTOMREQUEST, "GET");
             // receive server response ...
             curl_setopt($http, CURLOPT_RETURNTRANSFER, true);
-            $response = curl_exec($http);
+            // set the cURL time out
+            curl_setopt($http, CURLOPT_TIMEOUT_MS, $timeout);
+
+            if(array_key_exists("url",$proxy_details) && array_key_exists("port",$proxy_details)){
+                if($proxy_details['url'] != '' && $proxy_details['port'] != '') {
+
+                    // Set the proxy IP
+                    curl_setopt($http, CURLOPT_PROXY, $proxy_details['url']);
+                    // Set the port
+                    curl_setopt($http, CURLOPT_PROXYPORT, $proxy_details['port']);
+                    
+                }
+            }
+            if(array_key_exists("username",$proxy_details) && array_key_exists("password",$proxy_details)){
```

Comment:
Can we move this condition in url exist condition.
This will avoid adding user and password if url for proxy is not specified.